lib/web_ui/lib/src/engine/bitmap_canvas.dart

```diff
@@ -568,30 +579,76 @@ class BitmapCanvas extends EngineCanvas with SaveStackTracking {
   void drawImage(ui.Image image, ui.Offset p, ui.PaintData paint) {
     _applyPaint(paint);
     final HtmlImage htmlImage = image;
-    final html.Element imgElement = htmlImage.imgElement.clone(true);
+    final html.Element imgElement = htmlImage.cloneImageElement();
+    _drawImage(imgElement, p);
+    _childOverdraw = true;
+  }
+
+  void _drawImage(html.ImageElement imgElement, ui.Offset p) {
     imgElement.style
-      ..position = 'absolute'
-      ..transform = 'translate(${p.dx}px, ${p.dy}px)';
-    rootElement.append(imgElement);
+      ..position = 'absolute';
+    if (isClipped) {
+      final List<html.Element> clipElements =
+      _clipContent(_clipStack, imgElement, p, currentTransform);
```

Comment:
Please see _clipContent implementation. There was a TODO to reduce/merge multiple clips which should be addressed separately. We need benchmarks to make sure it is not too expensive since some are complex svg paths.